eng/scripts/Language-Settings.ps1

```diff
@@ -1,41 +1,52 @@
 $Language = "go"
 $packagePattern = "go.mod"
-# rewrite from ChangeLog-Operations.ps1 used in Get-ChangeLogEntriesFromContent for go uses vx.x.x as version number
-$RELEASE_TITLE_REGEX = "(?<releaseNoteTitle>^\#+\s+(?<version>v$([AzureEngSemanticVersion]::SEMVER_REGEX))(\s+(?<releaseStatus>\(.+\))))"
+$LanguageDisplayName = "go"
+$VERSION_LINE_REGEX = ".+\s*=\s*`".*v(?<version>$([AzureEngSemanticVersion]::SEMVER_REGEX))`""
 
-# rewrite from artifact-metadata-parsing.ps1 used in RetrievePackages for fetch go single module info
-function Get-go-PackageInfoFromPackageFile ($pkg, $workingDirectory)
+function Get-GoPackageId ($pkgDirectory) 
 {
-    $workFolder = $pkg.Directory
-    $releaseNotes = ""
+    $serviceDir = ""
+    $pkgId = ""
+    $sdkType = ""
 
-    if ($workFolder -match "sdk[\/|\\]((?<repopath>(?<service>.*?)[\/|\\])?(?<arm>arm)?(?<pkgname>.*))")
+    if ($pkgDirectory -match "sdk[\/|\\]((?<repopath>(?<service>.*?)[\/|\\])?(?<arm>arm)?(?<pkgname>.*))")
     {
         if ($matches["arm"])
         {
             $packageName = "arm" + $matches["pkgname"]
-            $rpName = $matches["service"]
-            $pkgId = "sdk/$rpName/$packageName"
+            $serviceDir = $matches["service"]
+            $pkgId = "sdk/$serviceDir/$packageName"
+            $sdkType = "mgmt"
         }
         else
         {
             $packageName = $matches["pkgname"]
+            $serviceDir = $packageName
```

Comment:
I think this is making too many assumptions. I think we still need to support cases where `$matches["service"]` is empty or not empty.

Reply:
Actually if we go witht he way ServiceDirectory is specified in the ci.yml this is how it should be.
For arm packages e.g. armagrifood the ServiceDirectory is `agrifood/armagrifood` for client packages e.g. azcore the ServiceDirectory is `azcore`.